corrections and updates to version 1.0

GlennBullingham opened the pull request:
Various updates and corrections for Version 1.0 documentation

GlennBullingham committed:
docs(readme): update repo list

- tidies style
- updates repo list
- clarifies OpenEBS / Mayastor positioning

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
fix(cluster-prep): only a msn can mount volumes

- corrects false assertion that non-MSN worker nodes can mount Mayastor volumes
- Style / clarity improvements

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
docs(deploy): update release tag in kubectl cmds

- replaces HEAD of master with the v1.0.2 tag, as the repo branching structure
  may be revised as of the release of v2.0, when master may no longer exist.

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
docs(test-app): local sc parameter corrections

- corrects the statements regarding the local parameter of StorageClass,
  providing direction to the user that 'local', with its associated
  behaviour is the only supported setting in this release

- updates master branch reference to tag value (v1.0.2) as this branch
  is due to be deprecated

- minor style / formatting corrections

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
docs(faqs): add testing details

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
docs(known-limit): remove reference to SC page

- removes the reference and link to the StorageClass explanation
  since the use of the "local" parameter with a value of true
  is mandatory in the v1.x release line

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
docs(known-issue): remove helm chart issue

- remove the incorrect Helm chart issue as this is resolved
  in the latest release

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
docs(sc-parameters): update local param details

- corrections to the 'local' parameter details
- adds mandate for this param value to be 'true' in this release

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
docs(pre-req): update min kernel version

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

tiagolobocastro commented on README.md:
```diff
@@ -1,31 +1,35 @@
 ---
 
-# Welcome to Mayastor!
+# Welcome to the Mayastor!
```
Is `the` necessary? Eg: [Welcome to Github](https://github.com/github/welcome-to-github)

GlennBullingham replied:
Absolutely not.  It's a typo, which I'll correct.  Good spot.

tiagolobocastro commented on README.md:
```diff
@@ -1,31 +1,35 @@
+- [openebs/mayastor](https://github.com/openebs/mayastor) : contains the source code of the data plane components
+- [openebs/mayastor-control-plane](https://github.com/openebs/mayastor-control-plane) : contains the source code of the control plane components
+- [openebs/mayastor-api](https://github.com/openebs/mayastor-api) : contains common protocol buffer definitions and OpenAPI specifications for Mayastor components
+- [openebs/mayastor-extensions](https://github.com/openebs/mayastor-extensions): contains source code which extends the functionality of the core control and data plane components, including the Helm chart for installation
```
We shouldn't reference the chart here since it's only valid for 2.0?

GlennBullingham replied:
Good point.  I'll remove that reference.

tiagolobocastro commented on quickstart/preparing-the-cluster.md:
````diff
@@ -22,7 +28,7 @@ HugePages_Surp:        0
 ```
 
-If fewer than 1024 pages are available then the page count should be reconfigured as required, accounting for any other workloads which may be co-resident on the worker node and which also require them.  For example:
+If fewer than 1024 pages are available then the page count should be reconfigured on the worker nodeas as required, accounting for any other workloads which may be scheduled on the same node and which also require them.  For example:
````
"node as" <- space

GlennBullingham replied:
Will fix

tiagolobocastro commented on quickstart/prerequisites.md:
```diff
@@ -4,16 +4,13 @@
-
-* Linux kernel **5.4 or higher** with the following modules loaded:
+* **x86-64** CPU cores with SSE4.2 instruction support
+* Linux kernel **5.13 or higher** with the following modules loaded:
```
Any reason to not list out the processor models from intel/amd?

GlennBullingham replied:
Advise from Nick w.r.t. being too specific.  As in, are we _certain_ that those families (and those alone) were the start point for that instruction set version?  If we're clear on the SSE version we require, just name that and let the user make the appropriate distinction

tiagolobocastro commented on reference/storage-class-parameters.md:
```diff
@@ -16,23 +16,17 @@ The setting is supported only when using "nvmf" protocol.
-{% hint style="info" %}
-Note that in order to enable the locality feature, the `volumeBindingMode` in storage class must be set to `WaitForFirstConsumer`.
+{% hint style="warning" %}
+This value must be set to true for correct operation of Mayastor provisioning and publishing.  It is recommended that  the `volumeBindingMode` in storage class be set to `WaitForFirstConsumer`.  This limitation will be removed in a future release
```
Right so now we're saying that it must be set to true, so we don't need to explain what happens with false, and we'll "fix" this anyway on the next release?

GlennBullingham replied:
That's my intention, based on conversations I believe we've had.  Do you concur?

tiagolobocastro replied:
We've had an issue raised recently where someone was trying to use local false; they expected to stand a volume on a non-engine node which currently is not supported even with local false.

> Dr. Harris, do you concur? 

I do :)

GlennBullingham replied:
Then we have consensus.  :-)

GlennBullingham committed:
fix(readme): remove ref to extensions repo

- removes mayastor-extensions repo from the list of repos
  associated with the project as this isn't used until
  version 2.0

- corrects a typo in the page title

- (addresses review comments)

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
fix(cluster-prep): corrects a typo

- addresses review comment

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

avishnu commented on README.md:
```diff
@@ -1,31 +1,34 @@
 ---
-description: 'The native NVMe-oF CAS engine for OpenEBS, by OpenEBS'
+description: 'The native NVMe-oF CAS engine for OpenEBS'
```
Should it be "of OpenEBS" instead of "for OpenEBS" ?

GlennBullingham replied:
I think you're correct.  Will change.

GlennBullingham replied:
Resolved: 3ca46b8

avishnu commented on README.md:
```diff
@@ -1,31 +1,34 @@
+- [openebs/mayastor](https://github.com/openebs/mayastor) : contains the source code of the data plane components
+- [openebs/mayastor-control-plane](https://github.com/openebs/mayastor-control-plane) : contains the source code of the control plane components
+- [openebs/mayastor-api](https://github.com/openebs/mayastor-api) : contains common protocol buffer definitions and OpenAPI specifications for Mayastor components
+- [openebs/mayastor-dependencies]:(https://github.com/openebs/mayastor-dependencies): contains dependencies common to the control and data plane repositories
```
Colon not needed after ]
Space needed after )

GlennBullingham replied:
Will fix

GlennBullingham replied:
Resolved: 3ca46b8

avishnu commented on quickstart/deploy-a-test-application.md:
```diff
@@ -50,12 +50,16 @@ If you used the storage class example from previous stage, then volume binding m
+The Mayastor CSI driver will cause the application pod and the corresponding Mayastor volume's NVMe target/controller ("Nexus") to be scheduled on the _same_ Mayastor Node, in order to assist with restoration of volume and application availabilty in the event of a storage node failure.
+
+{% hint style="warning" %}
+In this version, applications using PVs provisioned by Mayastor can only be successfully scheduled on Mayastor Nodes.  This behaviour is controlled by the `local:` parameter of the corresponding StorageClass, which by default is set to a value of `true`.  Therefore, this is the only supported value for this release - setting a non-local configuration may cause scheduling of the application pod to ...
```
Are there extra spaces after full-stops?

GlennBullingham replied:
https://www.cultofpedagogy.com/two-spaces-after-period/  :)

GlennBullingham replied:
Will adopt this style in future text.  At this point, double sentence spacing is already used throughout this page and the documentation in general.

anupriya0703 commented on quickstart/deploy-mayastor.md:
```diff
@@ -8,7 +8,7 @@ Before deploying and using Mayastor please consult the [Known Issues](https://ma
 In this Quickstart guide we demonstrate deploying Mayastor by using the Kubernetes manifest files provided within the project's repositories ([control plane components](https://github.com/openebs/mayastor-control-plane), [data plane components](https://github.com/openebs/mayastor)). These repositories are configured for the GitFlow release model, wherein the master branch contains official releas ...
 
-The steps and commands which follow are intended only for use with, and tested against, the latest release. Earlier releases or development versions may require a modified or different installation process.
+The steps and commands which follow are intended only for use with, and have only be tested against, the _latest release for version 1.x_. Earlier releases or development versions may require a modified or different installation process and _will_ require that the namifest files appropriate to that specific release are used.  These are available in the repositories named above - use the files fro ...
```
Typo (namifest -> manifest)

GlennBullingham replied:
Thanks.  Will fix.

GlennBullingham replied:
Resolved: 0aeb2e1

GlennBullingham committed:
fix(readme): correct typos

- corrects grammar in description field
- corrects typo in repo list syntax

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
fix(deploy-mayastor): correct a typo

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

avishnu commented on quickstart/deploy-a-test-application.md:
```diff
@@ -158,7 +162,7 @@ ID                                    REPLICAS  TARGET-NODE                ACCES
 
 {% hint style="info" %}
-Note: The FIO pod resource declaration from the Mayastor master branch references a PVC named `ms-volume-claim`, consistent with the example PVC created in this section of the quickstart. If you have elected to name your PVC differently, deploy the Pod using the example YAML, modifying the `claimName`field appropriately.
+Note: The example FIO pod resource declaration included with this release Mayastor master references a PVC named `ms-volume-claim`, consistent with the example PVC created in this section of the quickstart. If you have elected to name your PVC differently, deploy the Pod using the example YAML, modifying the `claimName` field appropriately.
```
Is "Mayastor master" needed?

GlennBullingham replied:
Absolutely not.  Will Fix.

GlennBullingham replied:
Resolved: 0b0574a

avishnu commented on quickstart/known-limitations.md:
```diff
@@ -12,7 +12,7 @@ Mayastor has no snapshot or cloning capabilities.
 
 Mayastor Volumes can be configured \(or subsequently re-configured\) to be composed of 2 or more "children" or "replicas"; causing synchronously mirrored copies of the volumes's data to be maintained on more than one worker node and Disk Pool. This contributes additional "durability" at the persistence layer, ensuring that viable copyies of a volume's data remain even if a Disk Pool device is los ...
```
Typo - copies

GlennBullingham replied:
Resolved: ec9b239

GlennBullingham committed:
fix(known-limit): fix typo

- corrects a typo
- improves clarifity of the 'durability' issue by removing extraneous info

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

GlennBullingham committed:
fix(deploy-test-app): corrects a typo

Signed-off-by: GlennBullingham <glenn.bullingham@datacore.com>

avishnu reviewed (approved):
LGTM !!
